src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java

```diff
@@ -178,10 +183,40 @@ public KubernetesComputer createComputer() {
         return new KubernetesComputer(this);
     }
 
+    public PodRetention getPodRetention(KubernetesCloud cloud) {
+        PodRetention retentionPolicy = cloud.getPodRetention();
+        if (template != null && template.getPodRetention() != PodRetention.DEFAULT) {
+            retentionPolicy = template.getPodRetention();
+        }
+        return retentionPolicy;
```

Comment:
Pod retention policy can be overridden by the pod template. Defaults to the cloud-level setting.